providers/redfish/firmware.go

```diff
@@ -409,6 +417,24 @@ func (c *Conn) FirmwareInstallStatus(ctx context.Context, installVersion, compon
 	switch {
 	case strings.Contains(vendor, constants.Dell):
 		task, err = c.dellJobAsRedfishTask(taskID)
+	case strings.Contains(vendor, constants.Packet) || strings.Contains(vendor, constants.Equinix):
```

Comment:
lets move this into `tasks.go` and make it more generic method that returns a `*gofish.Task` object,  it can replace the `openbmcGetTask()` method.  
We also don't need to check the vendor, and can just have a default switch case - since this is standard redfish

Reply:
Ok for the default, but we still need to check the vendor, as Dell doesn't use the default